### PR TITLE
Use imbu class

### PR DESCRIPTION
**Pull request by oxtopus**
requires https://github.com/numenta/nupic.research/pull/421

### CODE REVIEW

**Comment by oxtopus** [comment]
@BoltzmannBrain I'm experiencing the following error.  Any insight would be appreciated.

```
[INFO] Starting new HTTP connection (1): api.cortical.io
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/web.py-0.37-py2.7.egg/web/application.py", line 239, in process
    return self.handle()
  File "/usr/local/lib/python2.7/dist-packages/web.py-0.37-py2.7.egg/web/application.py", line 230, in handle
    return self._delegate(fn, self.fvars, args)
  File "/usr/local/lib/python2.7/dist-packages/web.py-0.37-py2.7.egg/web/application.py", line 420, in _delegate
    return handle_class(cls)
  File "/usr/local/lib/python2.7/dist-packages/web.py-0.37-py2.7.egg/web/application.py", line 396, in handle_class
    return tocall(*args)
  File "engine/fluent_api.py", line 168, in POST
    response = g_fluent.query(modelName, data)
  File "engine/fluent_api.py", line 118, in query
    sortedDistances = imbu.query(g_models[model], text)
  File "/opt/numenta/nupic.research/htmresearch/frameworks/nlp/imbu.py", line 227, in query
    sortResults=sortResults)
  File "/opt/numenta/nupic.research/htmresearch/frameworks/nlp/classification_model.py", line 176, in inferDocument
    return self._inferDocumentDetailed(tokenList, sortResults=sortResults)
  File "/opt/numenta/nupic.research/htmresearch/frameworks/nlp/classification_model.py", line 382, in _inferDocumentDetailed
    sortResults=False)
  File "/opt/numenta/nupic.research/htmresearch/frameworks/nlp/classify_fingerprint.py", line 145, in inferToken
    idList = [self.classifier.getPartitionId(i) for i in xrange(len(dist))]
AttributeError: 'KNNClassifier' object has no attribute 'getPartitionId'

[pid: 30|app: 0|req: 6/6] 192.168.99.1 () {46 vars in 819 bytes} [Mon Jan 18 22:02:46 2016] POST /fluent/CioWordFingerprint => generated 56362 bytes in 550 msecs (HTTP/1.1 500) 7 headers in 319 bytes (2 switches on core 0)
```

**Comment by BoltzmannBrain** [comment]
`getPartitionId()` is new in NuPIC, not included in released version 0.3.6, which is why the requirements doc specifies 0.3.7.dev0

**Comment by oxtopus** [comment]
Turns out docker isn't re-pulling the numenta/nupic docker image when I rebuild the imbu image and the sha in requirements.txt is ignored because the stated versions match. I was running against an older version of nupic, and fixed it by `docker pull`ing numenta/nupic. :rage: 

Thanks @lscheinkman!

**Comment by oxtopus** [comment]
Should be good as of 5911e6fe812d0e234768ebb72eae4d26a2d28834 and ready for review

File: imbu/destructive-build-and-refresh-container.sh [hunk 1]
```diff
@@ -3,11 +3,13 @@
...
 docker kill imbu
 docker rm imbu
-docker build -t imbu:latest .
+docker build --no-cache -t imbu:latest .
 docker run \
   --name imbu \
   -d \
   -p 8080:80 \
+  -e IMBU_LOAD_PATH_PREFIX=${IMBU_LOAD_PATH_PREFIX} \
+  -e IMBU_SAVE_PATH_PREFIX=${IMBU_SAVE_PATH_PREFIX} \
```
Comment (BoltzmannBrain) [inline imbu/destructive-build-and-refresh-container.sh]: Where is this used? It's also not included in the readme.

Comment (oxtopus) [inline imbu/destructive-build-and-refresh-container.sh]: It's not.  Removing now.

**Pull request merged**